src/quicktype-graphql-input/index.ts

```diff
@@ -357,6 +401,46 @@ class GQLQuery {
 
         return panic(`Unknown query operation type: "${query.operation}"`);
     }
+
+    makeInputType(builder: TypeBuilder, query: OperationDefinitionNode, queryName: string): TypeRef {
+        const name = queryName + 'Input';
+        const defs = query.variableDefinitions || [];
+        const properties = defs.reduce(
+            (props, definition) => {
+                let variableType = definition.type;
+                let optional = true;
+                let list = false;
+                if (variableType.kind === VariableKind.NON_NULL) {
+                    optional = false;
+                    variableType = variableType.type;
+                }
+                if (variableType.kind === VariableKind.LIST) {
+                    list = true;
+                    variableType = variableType.type;
+                }
+                if (variableType.kind !== VariableKind.NAMED) {
+                    return panic(`Named type not found for variable "${definition.variable.name.value}"`);
+                }
+                const gqlType = this._schema.types[variableType.name.value];
+                let irType = this.makeIRTypeFromFieldNode(
```

Comment:
I (misuse?) reuse this method in a few places since the parsing logic is the same for input and output types, except that 
1. input types can't refer to output types and vice versa
2. input types don't use selection sets
I stubbed out some of the arguments since they exist for the output types and selection set logic which should never trigger